rialto_airflow/snapshot.py

```diff
@@ -11,16 +13,81 @@ class Snapshot:
     """
 
     path: Path
-    time: str
+    timestamp: str
     database_name: str
-    authors_csv: Path
 
-    def __init__(self, path, database_name=None):
-        now = datetime.datetime.now()
-        self.time = now.strftime("%Y%m%d%H%M%S")
+    @classmethod
+    def create(cls, data_dir: Path, database_name: str | None = None):
+        """
+        Create a Snapshot in the given root data directory. Optionally you can
+        override the database name, which is really only useful in testing.
+        """
+        timestamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
 
-        self.path = Path(path) / "snapshots" / self.time
-        self.path.mkdir(parents=True)
+        path = data_dir / "snapshots" / timestamp
+        path.mkdir(parents=True, exist_ok=True)
 
-        self.database_name = database_name or f"rialto_{self.time}"
-        self.authors_csv = self.path / "authors.csv"
+        database_name = database_name or f"rialto_{timestamp}"
+
+        return cls(path, timestamp, database_name)
+
+    @classmethod
+    def get_latest(cls, data_dir: Path):
+        """
+        Returns the most recent completed Snapshot.
+        """
+        data_dir = data_dir
+
+        snapshots_dir = data_dir / "snapshots"
+        if not snapshots_dir.is_dir():
+            return None
+
+        snapshot_names = [p.name for p in snapshots_dir.iterdir()]
+
+        for snapshot_name in reversed(sorted(snapshot_names)):
+            # ignore things that might be here that aren't the right format
```

Comment:
I guess sorting by folder name is sufficient to get the late time stamped one

Reply:
I wanted to get them in reverse order, so the oldest was picked first. I can add a comment about this if it doesn't make sense?

Reply:
I understood the reverse part, was more of a question about if the way Python does sorting of arrays works the way we'd expect with the way snapshot folders are named ... at first thought I guess it does (e.g. `20250512193014` etc. always has bigger names being later in time?)

Reply:
Yes, I think that's how it should work. The initial list on line 45 could be in any order. Then the list is sorted, and then reversed, so we'd get them in descending order: 

```python
['20250512193014', '20250511123033', '20250411082119', ...]
```